src/lib.rs

```diff
@@ -289,6 +289,62 @@ impl<'a> DoubleArrayTrieBuilder<'a> {
     }
 }
 
+pub struct PrefixIter<'a> {
+    key_len: usize,
+    da: &'a DoubleArrayTrie,
+    char_indices: str::CharIndices<'a>,
+    b: i32,
+    n: i32,
+    p: usize,
+    reach_leaf: bool,
+}
+
+impl<'a> Iterator for PrefixIter<'a> {
+    type Item = (usize, usize);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.reach_leaf {
+            return None;
+        }
+        loop {
+            if let Some((i, c)) = self.char_indices.next() {
+                self.p = self.b as usize;
+                self.n = self.da.base[self.p];
+
+                if self.b == self.da.check[self.p] as i32 && self.n < 0 {
+                    self.p = self.b as usize + c as usize + 1;
+                    if self.b == self.da.check[self.p] as i32 {
+                        self.b = self.da.base[self.p];
+                    } else {
+                        self.reach_leaf = true;
+                    }
+
+                    return Some((i, (-self.n - 1) as usize));
+                }
+
```

Comment:
This is a buggy implementation.
This original implementation pushes new value(here we return Some), then the following 5 lines of code will be executed. Then, this loop ends.
Your implementation will not run the following code if above returns Some.

To fix this, save ``c`` in this ``Iterator`` using ``Option<char>``.

Reply:
Have no idea of a failed test case. 🤣 
Currently, all tests pass.